libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/alert/AlertView.java

```diff
@@ -62,6 +62,7 @@ protected void onFinishInflate() {
   protected void onDetachedFromWindow() {
     super.onDetachedFromWindow();
     if (countdownAnimation != null) {
+      countdownAnimation.removeAllListeners();
```

Comment:
I've tested this and it's not being called because `countdownAnimation` is `null` when `onDetachedFromWindow` is called - causing the leak.

I believe best approach here is bring the `WeakReference` back and remove `onDetachedFromWindow` code.